# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1523111833 14400
#      Sat Apr 07 10:37:13 2018 -0400
# Node ID dfc6ccc2a3e88dc30e62e39c14d9aac6adc410c9
# Parent  33bb7ffc54a1129a9c9efa5084d1fcbe697d3b30
set dev branch version to 5.0.0

New release version scheme follows that of GCC.  Briefly:

  * Any version X.0.0 means "this is an experimental development
    version".

  * Any version X.Y.1 means, "this is a pre-release version meant
    for bug fixing and testing".

  * Any version X.Y.0 with Y != 0 means "this is a released version".

More info here:

http://lists.gnu.org/archive/html/octave-maintainers/2018-03/msg00214.html

* configure.ac (AC_INIT): Set version to 5.0.0.
(OCTAVE_MAJOR_VERSION): Now 5.
(OCTAVE_MAJOR_VERSION, OCTAVE_PATCH_VERSION): Now 0.
(OCTAVE_RELEASE_DATE): Set to today; may update after 4.4.0 is released.

* chop.m, desktop.m, java2mat.m, tmpnam.m, toascii.m, bitmax.m:
Removal version is now 6, not 4.8.

* deprecate-props.tst: Update removal version numbers.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -15,34 +15,34 @@ dnl Copyright (C) 1993-2018 John W. Eato
 ### GNU General Public License for more details.
 ###
 ### You should have received a copy of the GNU General Public License
 ### along with Octave; see the file COPYING.  If not, see
 ### <https://www.gnu.org/licenses/>.
 
 ### Initialize Autoconf
 AC_PREREQ([2.65])
-AC_INIT([GNU Octave], [4.5.0+], [https://octave.org/bugs.html], [octave],
+AC_INIT([GNU Octave], [5.0.0], [https://octave.org/bugs.html], [octave],
         [https://www.gnu.org/software/octave/])
 
 ### Declare version numbers
 
 dnl Note that the version number is duplicated here and in AC_INIT because
 dnl AC_INIT requires it to be static, not computed from shell variables.
 
-OCTAVE_MAJOR_VERSION=4
-OCTAVE_MINOR_VERSION=5
-OCTAVE_PATCH_VERSION=0+
+OCTAVE_MAJOR_VERSION=5
+OCTAVE_MINOR_VERSION=0
+OCTAVE_PATCH_VERSION=0
 
 dnl PACKAGE_VERSION is set by the AC_INIT VERSION argument.
 OCTAVE_VERSION="$PACKAGE_VERSION"
 
 OCTAVE_COPYRIGHT="Copyright (C) 2018 John W. Eaton and others."
 
-OCTAVE_RELEASE_DATE="2016-11-13"
+OCTAVE_RELEASE_DATE="2018-04-07"
 
 ## The "API version" is used as a way of checking that interfaces in the
 ## liboctave and libinterp libraries haven't changed in a backwardly
 ## incompatible way when loading .oct files.  A better way to do this is with
 ## library versioning, but not all systems support it.
 ##
 ## NOTE: This macro will be removed in a future version of Octave!
 ## If you insist on checking for features using a version number, use the
diff --git a/scripts/deprecated/bitmax.m b/scripts/deprecated/bitmax.m
deleted file mode 100644
--- a/scripts/deprecated/bitmax.m
+++ /dev/null
@@ -1,52 +0,0 @@
-## Copyright (C) 2004-2018 John W. Eaton
-##
-## This file is part of Octave.
-##
-## Octave is free software: you can redistribute it and/or modify it
-## under the terms of the GNU General Public License as published by
-## the Free Software Foundation, either version 3 of the License, or
-## (at your option) any later version.
-##
-## Octave is distributed in the hope that it will be useful, but
-## WITHOUT ANY WARRANTY; without even the implied warranty of
-## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-## GNU General Public License for more details.
-##
-## You should have received a copy of the GNU General Public License
-## along with Octave; see the file COPYING.  If not, see
-## <https://www.gnu.org/licenses/>.
-
-## -*- texinfo -*-
-## @deftypefn {} {@var{r} =} bitmax (@var{precision})
-##
-## @code{bitmax} is deprecated and will be removed in Octave version 4.6.
-## Use @code{flintmax (precision) - 1} for the equivalent functionality.
-##
-## Return the largest integer @var{r} that can be represented within a
-## floating point value.
-##
-## The default class is @qcode{"double"}, but @qcode{"single"} is a valid
-## option.  On IEEE 754 compatible systems, @code{bitmax} is
-## @w{@math{2^{53} - 1}} for @qcode{"double"} and @w{@math{2^{24} - 1}} for
-## @qcode{"single"}.
-##
-## @seealso{flintmax, intmax, realmax, realmin}
-## @end deftypefn
-
-## Deprecated in version 4.2
-
-function r = bitmax (precision)
-
-  persistent warned = false;
-  if (! warned)
-    warned = true;
-    warning ("Octave:deprecated-function",
-             "bitmax is obsolete and will be removed from a future version of Octave, please use flintmax instead");
-  endif
-
-  if (nargin == 0)
-    precision = "double";
-  endif
-  r = flintmax (precision) - 1;
-
-endfunction
diff --git a/scripts/deprecated/chop.m b/scripts/deprecated/chop.m
--- a/scripts/deprecated/chop.m
+++ b/scripts/deprecated/chop.m
@@ -14,17 +14,17 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {} {} chop (@var{x}, @var{ndigits}, @var{base})
 ##
-## @code{chop} is deprecated and will be removed in Octave version 4.8.
+## @code{chop} is deprecated and will be removed in Octave version 6.
 ##
 ## Truncate elements of @var{x} to a length of @var{ndigits} such that the
 ## resulting numbers are exactly divisible by @var{base}.
 ##
 ## If @var{base} is not specified it defaults to 10.
 ##
 ## @example
 ## @group
diff --git a/scripts/deprecated/desktop.m b/scripts/deprecated/desktop.m
--- a/scripts/deprecated/desktop.m
+++ b/scripts/deprecated/desktop.m
@@ -14,17 +14,17 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {} {@var{used} =} desktop ("-inuse")
 ##
-## @code{desktop} is deprecated and will be removed in Octave version 4.8.
+## @code{desktop} is deprecated and will be removed in Octave version 6.
 ## Use @code{isguirunning} for the equivalent functionality.
 ##
 ## Return true if the desktop (GUI) is currently in use.
 ## @seealso{isguirunning}
 ## @end deftypefn
 
 ## Deprecated in version 4.4
 
diff --git a/scripts/deprecated/java2mat.m b/scripts/deprecated/java2mat.m
--- a/scripts/deprecated/java2mat.m
+++ b/scripts/deprecated/java2mat.m
@@ -13,17 +13,17 @@
 ## GNU General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {} {} java2mat (@var{javaobj})
-## @code{java2mat} is deprecated and will be removed in Octave version 4.8.
+## @code{java2mat} is deprecated and will be removed in Octave version 6.
 ##
 ## For the next two releases, use @code{__java2mat__} if necessary, and file
 ## a bug report explaining your programming use of @code{java2mat} and how it
 ## can't be done with other Octave functions.
 ## @end deftypefn
 
 function retval = java2mat (varargin)
 
diff --git a/scripts/deprecated/tmpnam.m b/scripts/deprecated/tmpnam.m
--- a/scripts/deprecated/tmpnam.m
+++ b/scripts/deprecated/tmpnam.m
@@ -15,17 +15,17 @@
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {} {@var{fname} =} tmpnam ()
 ## @deftypefnx {} {@var{fname} =} tmpnam (@var{dir})
 ## @deftypefnx {} {@var{fname} =} tmpnam (@var{dir}, @var{prefix})
-## @code{tmpnam} is deprecated and will be removed in Octave version 4.8.
+## @code{tmpnam} is deprecated and will be removed in Octave version 6.
 ## Use @code{tempname} instead.
 ##
 ## Return a unique temporary filename as a string.
 ##
 ## If @var{prefix} is omitted, a value of @qcode{"oct-"} is used.
 ##
 ## If @var{dir} is also omitted, the default directory for temporary files
 ## (@code{P_tmpdir} is used.  If @var{dir} is provided, it must exist,
diff --git a/scripts/deprecated/toascii.m b/scripts/deprecated/toascii.m
--- a/scripts/deprecated/toascii.m
+++ b/scripts/deprecated/toascii.m
@@ -14,17 +14,17 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {} {} toascii (@var{str})
 ##
-## @code{toascii} is deprecated and will be removed in Octave version 4.8.
+## @code{toascii} is deprecated and will be removed in Octave version 6.
 ## Use @code{double} instead.
 ##
 ## Return ASCII representation of @var{str} in a matrix.
 ##
 ## For example:
 ##
 ## @example
 ## @group
diff --git a/test/deprecate-props.tst b/test/deprecate-props.tst
--- a/test/deprecate-props.tst
+++ b/test/deprecate-props.tst
@@ -29,67 +29,67 @@
 %!      error ("Please remove %s property %s", get (h, "type"), prop);
 %!    elseif (! isempty (val) && any (strcmp (val, set (h, prop))))
 %!      error ("Please remove '%s' from allowed values for %s property %s",
 %!             val, get (h, "type"), prop);
 %!    endif
 %!  endif
 %!endfunction
 
-## annotation rectangle "edgecolor" deprecated in 4.4, remove from 4.7.+
+## annotation rectangle "edgecolor" deprecated in 4.4, remove from 6.
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   ha = annotation ("rectangle");
-%!   testprop (ha, "edgecolor", "4.6");
+%!   testprop (ha, "edgecolor", "6.0");
 %! unwind_protect_cleanup
 %!   close (hf);
 %! end_unwind_protect
 
 ## figure "doublebuffer, mincolormap, wvisual, wvisualmode, xdisplay,
-## xvisual, xvisualmode" deprecated in 4.4, remove from 4.7.+
+## xvisual, xvisualmode" deprecated in 4.4, remove from 6.
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
-%!   testprop (hf, "doublebuffer", "4.6");
-%!   testprop (hf, "mincolormap", "4.6");
-%!   testprop (hf, "wvisual", "4.6");
-%!   testprop (hf, "wvisualmode", "4.6");
-%!   testprop (hf, "xdisplay", "4.6");
-%!   testprop (hf, "xvisual", "4.6");
-%!   testprop (hf, "xvisualmode", "4.6");
+%!   testprop (hf, "doublebuffer", "6.0");
+%!   testprop (hf, "mincolormap", "6.0");
+%!   testprop (hf, "wvisual", "6.0");
+%!   testprop (hf, "wvisualmode", "6.0");
+%!   testprop (hf, "xdisplay", "6.0");
+%!   testprop (hf, "xvisual", "6.0");
+%!   testprop (hf, "xvisualmode", "6.0");
 %! unwind_protect_cleanup
 %!   close (hf)
 %! end_unwind_protect
 
-## axes "drawmode" deprecated in 4.4, remove from 4.7.+
+## axes "drawmode" deprecated in 4.4, remove from 6.
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   hax = axes ();
-%!   testprop (hax, "drawmode", "4.6");
+%!   testprop (hax, "drawmode", "6.0");
 %! unwind_protect_cleanup
 %!   close (hf);
 %! end_unwind_protect
 
 ## text/uicontrol/uipanel/uibuttongroup  "demi" and "light" values for
-## "fontweight" property are deprecated in 4.4, remove from 4.7.+:
+## "fontweight" property are deprecated in 4.4, remove from 6.
 ##   * remove "demi" and "light" options in graphics.in.h,
 ##   QtHandlesUtils.cc and ft-text-renderer.cc
 ##   * remove warnings from update_fontweight in graphics.in.h
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   ht = text ();
-%!   testprop (ht, "fontweight", "4.6", "demi");
-%!   testprop (ht, "fontweight", "4.6", "light");
+%!   testprop (ht, "fontweight", "6.0", "demi");
+%!   testprop (ht, "fontweight", "6.0", "light");
 %!   hui = uicontrol ();
-%!   testprop (hui, "fontweight", "4.6", "demi");
-%!   testprop (hui, "fontweight", "4.6", "light");
+%!   testprop (hui, "fontweight", "6.0", "demi");
+%!   testprop (hui, "fontweight", "6.0", "light");
 %!   hui = uipanel ();
-%!   testprop (hui, "fontweight", "4.6", "demi");
-%!   testprop (hui, "fontweight", "4.6", "light");
+%!   testprop (hui, "fontweight", "6.0", "demi");
+%!   testprop (hui, "fontweight", "6.0", "light");
 %!   hui = uibuttongroup ();
-%!   testprop (hui, "fontweight", "4.6", "demi");
-%!   testprop (hui, "fontweight", "4.6", "light");
+%!   testprop (hui, "fontweight", "6.0", "demi");
+%!   testprop (hui, "fontweight", "6.0", "light");
 %! unwind_protect_cleanup
 %!   close (hf);
 %! end_unwind_protect
